### ISSUE TITLE
General Settings page doesn't save with Php 7.2 or above

### ISSUE THREAD

**Issue by Robbt** [LibreTime/libretime#737]
**Describe the bug**
See #707 for the error.
Basically if the logo is empty the form fails to validate because the zend_validate tries pass a null value as a parameter to the count function which is not OK according to the tighter type checking of 7.2 and above.

Basically there 7.2 compatibility issue in zf1s - I put a PR into fix it - see https://github.com/zf1s/zend-validate/pull/1

We will need to use the master-dev in composer for zend-validate once it is merged.

**Comment by paddatrapper** [comment]
Your PR has been merged into zf1s. So the next step is to point composer to zend-validate master?

**Comment by paddatrapper** [comment]
I see composer is already using master for zend-validate.

**Comment by Robbt** [comment]
Yeah this bug should be fixed if people update their composer to pull the latest zend-validate, nothing that we need to do so I'm going to mark it closed.

**Comment by maydo** [comment]
> 
> 
> Yeah this bug should be fixed if people update their composer to pull the latest zend-validate, nothing that we need to do so I'm going to mark it closed.

zf1s/zend-validate                dev-master 424607a 
i have still the issue on latest master
general settings does not save with empty img

**Comment by Robbt** [comment]
can you check your composer.json and see what it has for zend-validate ?

**Comment by maydo** [comment]
"zf1s/zend-validate": "dev-master@dev",

composer show
![20190303002640](https://user-images.githubusercontent.com/27486434/53688886-22f78880-3d4b-11e9-80f4-75fb9020ee75.jpg)

**Comment by Robbt** [comment]
Ok, so for some reason upon installation the latest versions aren't being installed and I needed to run composer update to get the newest version including the fix. So we need to figure out why the latest zf1s libraries aren't installed during the install process in the first place.

This is fixed upstream still but the latest version isn't downloaded by default evidently.

**Comment by maydo** [comment]
`Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 0 installs, 0 updates, 0 removals`

also if i run compose update its same, even with deleted composer.lock file

**Comment by Robbt** [comment]
yeah I ran composer update and it got the latest version. you should compare your version of the vendor/z1fs/zend-validate/Count.php file with the change I put above and see if you indeed have the latest version. I am doing this in a vagrant development environment. If you are on a standard install you will need to update the vendor files that are included with airtime_mvc. I'm not sure the exact difference. Looking into the inability to move files in the schedule currently as that is a bigger blocker.

**Comment by maydo** [comment]
i got it fixed, thank you

**Comment by Robbt** [comment]
great any steps for how you fixed it ?

**Comment by maydo** [comment]
stopped all services
re-run the installer
start services again

not sure but i think there was some issues from the previous tarball installation 3.0.0-alpha.6 installed on a fresh ubuntu 18.04 than uninstalled with uninstaller script, cloned master from git and installed again

**Comment by Robbt** [comment]
that makes sense. ok i'm going to close this again. but if this happens subsequently we can reopen.
